File: 066/cpp/main.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        std::vector<int> result;
        int r = 1;
        for (int i = digits.size() - 1; i > -1; i--) {
            r += digits[i];
            result.insert(result.begin(), r % 10);
            r /= 10;
        }
        if (r > 0) {
            result.insert(result.begin(), r);
        }
        return result;
    }
};

File: 066/rust/main.rs
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::new();
        let mut r = 1;
        for digit in digits.iter().rev() {
            r += digit;
            result.insert(0, r % 10);
            r /= 10;
        }
        if r > 0 {
            result.insert(0, r);
        }
        return result;
    }
}
